fineract-provider/src/main/java/org/apache/fineract/portfolio/loanaccount/domain/LoanAccountDomainServiceJpa.java

```diff
@@ -689,8 +689,6 @@ public LoanTransaction creditBalanceRefund(final Loan loan, final LocalDate tran
         loanRefundService.creditBalanceRefund(loan, newCreditBalanceRefundTransaction, existingTransactionIds,
                 existingReversedTransactionIds);
 
-        newCreditBalanceRefundTransaction = this.loanTransactionRepository.saveAndFlush(newCreditBalanceRefundTransaction);
```

Comment:
If this transaction is not flushed, in the `businessEventNotifierService
                .notifyPostBusinessEvent(new LoanCreditBalanceRefundPostBusinessEvent(newCreditBalanceRefundTransaction));` we wont have id...